Update doc about different configuration options (#161)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/USAGE.md

| Environment variable | Description | Default |
|-|-|-|

> | `OTEL_TRACE_CONFIG_FILE` | The file path of a JSON configuration file that will be loaded. | ```%WorkingDirectory%/datadog.json``` |

Reviewer: @dszmigielski this name here needs to be also updated.

> | `OTEL_TRACE_CONFIG_FILE` | Loads the json configuration file with configuration values. Overloaded value must be set via Environment variable. | ```{Environment.CurrentDirectory}/datadog.json``` |

Reviewer 2: How about `The file path of a JSON configuration file that will be loaded` ?

> | `OTEL_TRACE_CONFIG_FILE` | Loads the json configuration file with configuration values. Overloaded value must be set via Environment variable. | ```{Environment.CurrentDirectory}/datadog.json``` |

Reviewer 2: `{Environment.CurrentDirectory}/datadog.json` -> I suggest to use `CurrentWorkingDirectory` or just `WorkingDirectory` https://en.wikipedia.org/wiki/Working_directory

| `OTEL_PROFILER_PROCESSES` | Sets the filename of executables the profiler can attach to. If not defined (default), the profiler will attach to any process. Supports multiple values separated with comma, for example: `MyApp.exe,dotnet.exe` |  |
| `OTEL_PROFILER_EXCLUDE_PROCESSES` | Sets the filename of executables the profiler cannot attach to. If not defined (default), the profiler will attach to any process. Supports multiple values separated with comma, for example: `MyApp.exe,dotnet.exe` |  |

## Ways to configure

> There are 3 ways to apply configuration values (Priority is from first to last):
> 
> 1. Environment Variables
> 2. Web.config / App.config
> 3. Json configuration file

Reviewer 2: ```suggestion
There are the following ways to apply configuration values (priority is from first to last):

1. [Environment variables)(#environment-variables)
2. [`web.config` or `app.config` file](#web.config-and-app.config)
3. [JSON configuration file](#json-configuration-file)
```

### Environment variables

> Environment variables are the main way to configure values. After setting up environment variable, it can't be overridden by lower priority configuration option. 

Reviewer 2: ```suggestion
Environment variables are the main way to configure values.  A setting configured via an environment variable cannot be overridden.
```

> ### Web.config and app.config

Reviewer 2: ```suggestion
### web.config and app.config
```

> When your application is running on .NET Framework (Does not work with .NET Core or later), all unset configuration values can be set via web configuration file (web.config) or application configuration file (app.config).

Reviewer 2: ```suggestion
For an application is running on .NET Framework, web configuration file (`web.config`) or application configuration file (`app.config`) can be used to configure settings.
```

See example with `OTEL_SERVICE` overload.

```xml
<configuration>
  <appSettings>
    <add key="OTEL_SERVICE" value="my-service-name" />
  </appSettings>
</configuration>
```

> ### Json configuration file

Reviewer 2: ```suggestion
### JSON configuration file
```

> Json configuration file has the least priority. To use this option, all values that need to be overridden must be unset at first. By default if `OTEL_TRACE_CONFIG_FILE` is unset, the application is searching for `datadog.json` in current working directory (acquired by `Environment.CurrentDirectory`).

Reviewer 2: ```suggestion
By default, if `OTEL_TRACE_CONFIG_FILE` is unset, the application is searching for `datadog.json` in the current working directory (acquired by [`Environment.CurrentDirectory`](https://docs.microsoft.com/en-us/dotnet/api/system.environment.currentdirectory?view=net-5.0)).
```
